src/config.ts

```diff
@@ -0,0 +1,105 @@
+import {ShopifyError, UninitializedConfigError, PrivateAppError} from './error';
+import {SessionStorage} from './auth/session/session_storage';
+import {ApiVersion, ConfigParams} from './base-types';
+import {AuthScopes} from './auth/scopes';
+import {MemorySessionStorage} from './auth/session/storage/memory';
+
+interface ConfigInterface extends ConfigParams {
+  hostScheme: string;
+  sessionStorage: SessionStorage;
+  scopes: AuthScopes;
+}
+
+export const config: ConfigInterface = {
+  apiKey: '',
+  apiSecretKey: '',
+  scopes: new AuthScopes([]),
+  hostName: '',
+  hostScheme: 'https',
+  apiVersion: ApiVersion.Unstable,
+  isEmbeddedApp: true,
+  isPrivateApp: false,
+  // TS hack as sessionStorage is guaranteed to be set
+  // to a correct value in `initialize()`.
+  sessionStorage: null as unknown as SessionStorage,
+};
+
+export function setConfig(params: ConfigParams): void {
+  let scopes: AuthScopes;
+  if (params.scopes instanceof AuthScopes) {
+    scopes = params.scopes;
+  } else {
+    scopes = new AuthScopes(params.scopes);
+  }
+
+  // Make sure that the essential params actually have content in them
+  const missing: string[] = [];
+  if (!params.apiKey.length) {
+    missing.push('apiKey');
+  }
+  if (!params.apiSecretKey.length) {
+    missing.push('apiSecretKey');
+  }
+  if (!scopes.toArray().length) {
+    missing.push('scopes');
+  }
+  if (!params.hostName.length) {
+    missing.push('hostName');
+  }
+
+  if (missing.length) {
+    throw new ShopifyError(
+      `Cannot initialize Shopify API Library. Missing values for: ${missing.join(
+        ', ',
+      )}`,
+    );
+  }
```

Comment:
By rearranging this a bit and adding a `.length` property to `AuthScopes` (in case `param.scopes` is already that type on entry), this could be tidied up to be something like the following, which would allow relatively easy addition of other mandatory keys in future.
```suggestion
  // Make sure that the essential params actually have content in them
  const mandatory: string[]= ['apiKey', 'apiSecretKey', 'scopes', 'hostName'];
  const missing: string[] = [];
  keys.forEach((key) => { if (!params[key].length) { missing.push(key); }});

  if (missing.length) {
    throw new ShopifyError(
      `Cannot initialize Shopify API Library. Missing values for: ${missing.join(
        ', ',
      )}`,
    );
  }

  if (params.scopes instanceof AuthScopes) {
    config.scopes = params.scopes;
  } else {
    config.scopes = new AuthScopes(params.scopes);
  }
```